ql/src/java/org/apache/hadoop/hive/ql/optimizer/Optimizer.java

```diff
@@ -79,7 +79,7 @@ public void initialize(HiveConf hiveConf) {
         || postExecHooks.contains("org.apache.hadoop.hive.ql.hooks.PostExecutePrinter")
         || postExecHooks.contains("org.apache.hadoop.hive.ql.hooks.LineageLogger")
         || postExecHooks.contains("org.apache.atlas.hive.hook.HiveHook")) {
-      transformations.add(new Generator(postExecHooks));
+      transformations.add(Generator.fromConf(hiveConf));
```

Comment:
I assume that before that change, we read the transformations from the postExecHooks that is as far I remember, calculated once. And from now, we read those again at every single query execution. 
Am I correct?

Reply:
`postExecHooks` also calculated at every query execution:
https://github.com/apache/hive/blob/09553fca66ff69ff870c8a181750b70d81a8640e/ql/src/java/org/apache/hadoop/hive/ql/optimizer/Optimizer.java#L75-L77

I think this should be addressed in a follow-up only if this is bottleneck.

Reply:
thx